docs/src/proposals/accounts-db-replication.md

```diff
@@ -0,0 +1,189 @@
+and sends `ReplRpcAccountsResponse` to the requestor. The response contains the count of the
+ReplAccountInfo and the vector of ReplAccountInfo. This service runs both in the validator
+and the replica relaying replication information. The server can stream the account information
+from its AccountCache or from the storage if already flushed. This is similar to how a snapshot
+package is created from the AccountsDb with the difference that the storage does not need to be
+flushed to the disk before streaming to the client. If the account data is in the cache, it can
+be directly streamed. Care must be taken to avoid the account data for a slot get cleaned while
+serving the streaming. When attempting a replication of a slot, if the slot is already cleaned
+up with accounts data cleaned as result of update in later rooted slots, the replica should 
+forsake this slot and try the later uncleaned root slot.
+
+During replication we also need to replicate the information of accounts
+who have been cleaned up due to zero lamports, i.e. we need to be able to tell the difference
+for the account in a given slot: it is not updated and hence no storage entry in that slot and
+the accounts have 0 lamports and have been cleaned up through the history. We may record this
+via some "Tombstone" mechanism -- recording the dead accounts cleaned up for a slot. The
+tombstones themselves can be removed after exceeding retention period expressed as epochs.
+And attempt to replicate slots with tombstones removed will fail and the replica should skip
+this slot and try later ones.
+
+The `JsonRpcAccountsService`, this is the RPC service serving client requests for account
+information. The existing JsonRpcService serves other client calls than AccountsDb ones.
+The replica node only serves the AccountsDb calls.
+
+The existing JsonRpcService requires `BankForks`, `OptimisticallyConfirmedBank` and
+`BlockCommitmentCache` to load the Bank. The JsonRpcAccountsService will need to use 
+information obtained from ReplRpcUpdatedSlotsResponse to construct the AccountsDb. Question,
+do we still need the Bank in the replica?
+
+
+The `AccountsBackgroundService`, this service also runs in the replica which is responsible
+for taking snapshots periodically and shrinking the AccountsDb and doing accounts cleaning.
+The existing code also uses BankForks which we need to keep in the replica.
+
+### Compatibility Consideration
+
+For protocol compatiblilty considerations, all the requests have the replication version which is
+initially set to 1. Alternatively, we can use the validator's version. The RPC server side
+shall check the request version and fail if it is not supported.
+
+### Replication Setup
+To limit adverse effects on the validator and the replica due to replication, they can be configured
+with a list of replica nodes which can form a replication pair with it. And the replica node is
+configured with the validator which can serve its requests.
```

Comment:
This will just be one replica, right?

Reply:
Yes

Reply:
Let's update this section to reflect that, then.